actors/builtin/market/market_test.go

```diff
@@ -1302,7 +1302,34 @@ func TestCronTickTimedoutDeals(t *testing.T) {
 		require.Equal(t, big.Zero(), actor.getEscrowBalance(rt, provider))
 		require.Equal(t, big.Zero(), actor.getLockedBalance(rt, provider))
 
-		actor.assertDealDeleted(rt, dealId)
+		actor.assertDealDeleted(rt, dealId, d)
+	})
+
+	t.Run("publishing timed out deal again should work after cron tick as it should no longer be pending", func(t *testing.T) {
+		rt, actor := basicMarketSetup(t, owner, provider, worker, client)
+		dealId := actor.generateAndPublishDeal(rt, client, mAddrs, startEpoch, endEpoch)
+		d := actor.getDealProposal(rt, dealId)
+
+		// publishing will fail as it will be in pending
+		d2 := actor.generateDealAndAddFunds(rt, client, mAddrs, startEpoch, endEpoch)
+		params := mkPublishStorageParams(d2)
+		rt.ExpectValidateCallerType(builtin.AccountActorCodeID, builtin.MultisigActorCodeID)
+		rt.ExpectSend(provider, builtin.MethodsMiner.ControlAddresses, nil, abi.NewTokenAmount(0), &miner.GetControlAddressesReturn{Worker: worker, Owner: owner}, 0)
+		rt.SetCaller(worker, builtin.AccountActorCodeID)
+		rt.ExpectVerifySignature(crypto.Signature{}, d2.Client, mustCbor(&d2), nil)
+		rt.ExpectAbort(exitcode.ErrIllegalArgument, func() {
+			rt.Call(actor.PublishStorageDeals, params)
+		})
+		rt.Verify()
+
+		// do a cron tick for it -> should time out and get slashed
+		rt.SetEpoch(startEpoch)
+		rt.ExpectSend(builtin.BurntFundsActorAddr, builtin.MethodSend, nil, d.ProviderCollateral, nil, exitcode.Ok)
+		actor.cronTick(rt)
+		actor.assertDealDeleted(rt, dealId, d)
```

Comment:
The problem is that this method doesn't check `st.PendingProposals`, so this test passed even though it shouldn't have. (I think!)

Reply:
@wadeAlexC The test was passing because `msm.dealStates.Get(dealID)` in `CronTick` returns a state with `LastUpdatedEpoch = epochUndefined` if a state does not exist i.e. the deal has timed out.

However, for readability, I've fixed this by deleting pending timed-out deals as well.

Reply:
I see, didn't notice this behavior in the `Get` method. Thanks for pointing this out!